.github/workflows/ci.yml

```diff
@@ -52,8 +52,8 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - { name: "nvc++ 22.11", cuda: "11.8", cxx: "mpic++",  tag: "nvhpc22.11",       gpu: "v100", driver: "520", arch: "amd64" }
-          - { name: "clang 16", cuda: "11.8",    cxx: "clang++", tag: "llvmdev-cuda11.8", gpu: "v100", driver: "520", arch: "amd64" }
+          - { name: "nvc++ 23.1", cuda: "12.0",  cxx: "mpic++",  tag: "nvhpc23.1",       gpu: "v100", driver: "520", arch: "amd64" }
```

Comment:
To be sure shouldnt the driver version also change?

Reply:
Yeah, but I'll have to check if we have any 525 action runners yet. If not, we could disable the `NVIDIA_REQUIRE_CUDA` checks and `cuda-compat` might save us?

Reply:
Nevermind, I forgot `cuda-compat` doesn't apply to compilation. Will ask about our timelines for 525 runners.

Reply:
Yeah, we'll have to wait until we have action runners with driver 525.